src/spring-cloud/azext_spring_cloud/_utils.py

```diff
@@ -211,3 +211,12 @@ def _get_persistent_disk_size(tier):  # pylint: disable=too-many-return-statemen
     if tier == 'STANDARD':
         return 50
     return 50
+
+
+def get_portal_uri(cli_ctx):
+    """Get the Azure Portal URL in the current cloud."""
+    try:
+        return cli_ctx.cloud.endpoints.__getattribute__('portal')
+    except Exception as e:
+        logger.debug("Could not get Azure Portal endpoint. Exception: %s", str(e))
```

Comment:
You can use `cli_ctx.cloud.endpoints.portal` directly and catch `CloudEndpointNotSetException`.

Reply:
Partial updated. As the portal is just added in CLI core, `cli_ctx.cloud.endpoints.portal` against old cli version will throw `AttributeError` rather than `CloudEndpointNotSetException`